src/main/java/com/dsi/ecommerce/controller/ProductController.java

```diff
@@ -31,14 +32,14 @@
 
     @RequestMapping("/products")
 
-    public String getAllProducts(@AuthenticationPrincipal MyUserDetail principal, Model model) {
```

Comment:
What about if someone is logged in and hits `/products` ? His total cost will be ***0** in the navbar  and carts empty ?